### PR TITLE
Add installation instructions

### PR DESCRIPTION
**Pull request by skejserjensen**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,6 +18,33 @@ MiniModelarDB intentionally does not gather usage data. So, all users are highly
...
+
+#### macOS
+1. Install the Xcode Command Line Developer Tools: `xcode-select --install`
+2. Install [CMake](https://cmake.org/) and follow the _"How to Install For Command Line Use"_ menu item.
+
+#### Windows
+1. Install a supported version of [Visual Studio](https://visualstudio.microsoft.com/vs/older-downloads/) with Visual C++:\
+-- Visual Studio 2017 ([Supported](https://github.com/microsoft/snmalloc/blob/main/docs/BUILDING.md#building-on-windows))\
+-- Visual Studio 2019 ([Supported](https://github.com/microsoft/snmalloc/issues/525#issuecomment-1128901104))\
+-- Visual Studio 2022 (Unsupported)
```
[the remark this answers is not in the document]
Comment (skejserjensen) [reply README.md]: Fixed.

File: README.md [hunk 2]
```diff
@@ -18,6 +18,33 @@ MiniModelarDB intentionally does not gather usage data. So, all users are highly
...
+2. Install [CMake](https://cmake.org/) and select one of the following options during installation:\
+-- _Add CMake to the system PATH for all users_\
+-- _Add CMake to the system PATH for current user_
+
+### All
```
[the remark this answers is not in the document]
Comment (skejserjensen) [reply README.md:41]: Reduced the other headers to "###".

**Review by CGodiksen — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -18,8 +18,35 @@ MiniModelarDB intentionally does not gather usage data. So, all users are highly
 encouraged to post comments, suggestions, and bugs as GitHub issues, especially
 if a limitation of MiniModelarDB prevents it from being used in a particular domain.
 
+## Installation
```
Comment (chrthomsen) [inline README.md:21]: "Building" or "Installation"?

Does `cargo build` also place the executable in a specific directory and add configuration?

File: README.md [hunk 4]
```diff
@@ -30,8 +57,8 @@ All modules must have an accompanying doc comment that describes the general fun
 of the public functions, structs, enums, or other central elements of the module can be included.
 
 ### Testing
-All public and private functions must be covered by unit tests. Full coverage is required, which means all 
-branches of computation within the function needs to be tested. 
+All public and private functions must be covered by unit tests. Full coverage is required, which means all
+branches of computation within the function needs to be tested.
 
 ### Crates
 To avoid confusion and unnecessary dependencies, a list of utility crates are included. Note that this only includes crates
```
Comment (chrthomsen) [inline README.md:64]: Not sure I understand this. Why does that avoid confusion and unnecessary dependencies?

File: README.md [hunk 5]
```diff
@@ -18,8 +18,35 @@ MiniModelarDB intentionally does not gather usage data. So, all users are highly
...
+
+### macOS
+1. Install the Xcode Command Line Developer Tools: `xcode-select --install`
+2. Install [CMake](https://cmake.org/) and follow the _"How to Install For Command Line Use"_ menu item.
+
+### Windows
+1. Install a supported version of [Visual Studio](https://visualstudio.microsoft.com/vs/older-downloads/) with Visual C++:
+   * Visual Studio 2017 ([Supported](https://github.com/microsoft/snmalloc/blob/main/docs/BUILDING.md#building-on-windows))
+   * Visual Studio 2019 ([Supported](https://github.com/microsoft/snmalloc/issues/525#issuecomment-1128901104))
+   * Visual Studio 2022 (Unsupported)
```
Comment (chrthomsen) [inline README.md:36]: Why is is listed as the others (under supported versions) if is not supported?

Should it rather be "Note that Visual Studio 2022 is not supported"?

**Review by chrthomsen — COMMENTED** [review]
Looks good. I added some questions.

File: README.md (hunk 3 above)
Comment (skejserjensen) [reply README.md:21]: Kept **Installation** as the headline and instead extended **All** to also include how to run the system and how to move the binaries to whatever directory the user prefers.

File: README.md (hunk 4 above)
Comment (skejserjensen) [reply README.md:64]: The intention was to list the libraries the system use for specific functionality so different developers did not accidentally add multiple libraries that provide the same functionality. However, if that part of the README needs to be updated it should probably be in a different PR.

File: README.md (hunk 5 above)
Comment (skejserjensen) [reply README.md:36]: Fixed.

**Review by chrthomsen — APPROVED** [review]

**Pull request merged**